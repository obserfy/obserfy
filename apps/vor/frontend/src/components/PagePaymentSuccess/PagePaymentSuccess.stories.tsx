import { Meta, Story } from "@storybook/react"
import PagePaymentSuccess, {
  PagePaymentSuccessProps,
} from "./PagePaymentSuccess"

export default {
  title: "Core/PagePaymentSuccess",
  component: PagePaymentSuccess,
  parameters: {
    componentSubtitle: "Just a simple PagePaymentSuccess",
  },
} as Meta

const Template: Story<PagePaymentSuccessProps> = (args) => (
  <PagePaymentSuccess {...args} />
)

export const Default = Template.bind({})
Default.args = {}
