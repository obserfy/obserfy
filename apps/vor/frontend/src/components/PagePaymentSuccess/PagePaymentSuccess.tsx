import { t, Trans } from "@lingui/macro"
import { FC } from "react"
import { Box, Button, Card, Flex, Image } from "theme-ui"
import { useGetSchool } from "../../hooks/api/schools/useGetSchool"
import { ReactComponent as SuccessIllustration } from "../../images/success.svg"
import { ADMIN_SUBSCRIPTION_URL, ADMIN_URL } from "../../routes"
import { Link } from "../Link/Link"
import TopBar, { breadCrumb } from "../TopBar/TopBar"
import Typography from "../Typography/Typography"

export interface PagePaymentSuccessProps {}

const PagePaymentSuccess: FC<PagePaymentSuccessProps> = () => {
  useGetSchool()

  return (
    <Box mx="auto" sx={{ maxWidth: "maxWidth.xsm" }}>
      <TopBar
        breadcrumbs={[
          breadCrumb(t`Admin`, ADMIN_URL),
          breadCrumb(t`Plans & Billing`, ADMIN_SUBSCRIPTION_URL),
          breadCrumb(t`Payment Success `),
        ]}
      />

      <Card sx={{ borderRadius: [0, 32] }} py={5} mt={[0, 4]}>
        <Image as={SuccessIllustration} width="100%" px={[5, 6]} pb={4} />

        <Typography.H4 sx={{ textAlign: "center" }} mb={3}>
          <Trans>Thanks for Subscribing!</Trans>
        </Typography.H4>
        <Typography.Body
          sx={{ textAlign: "center", fontSize: 2 }}
          mb={4}
          px={4}
        >
          <Trans>
            Your purchase is complete. A receipt of your order will be sent to
            you by email. Thanks for supporting us!
          </Trans>
        </Typography.Body>

        <Flex sx={{ flexDirection: "column", alignItems: "center" }}>
          <Link to={ADMIN_SUBSCRIPTION_URL}>
            <Button
              p={3}
              sx={{ width: "100%", fontWeight: "bold", borderRadius: 16 }}
            >
              <Trans>Back to Billing</Trans>
            </Button>
          </Link>
        </Flex>
      </Card>
    </Box>
  )
}

export default PagePaymentSuccess
