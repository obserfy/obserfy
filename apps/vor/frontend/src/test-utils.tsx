/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
