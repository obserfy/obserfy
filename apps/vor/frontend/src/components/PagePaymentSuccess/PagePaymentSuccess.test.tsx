import { render } from "../../test-utils"
import PagePaymentSuccess from "./PagePaymentSuccess"

describe("PagePaymentSuccess", () => {
  it("should render correctly", () => {
    const { container } = render(<PagePaymentSuccess />)
    expect(container).toMatchSnapshot()
  })
})
