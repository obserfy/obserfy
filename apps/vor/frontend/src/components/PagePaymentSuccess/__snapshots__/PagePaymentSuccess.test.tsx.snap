// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePaymentSuccess should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 500px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-4:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-7 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-7:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-7:last-child>.splitter {
  display: none;
}

.emotion-8:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-13:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-13:last-child>.splitter {
  display: none;
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  padding-top: 64px;
  padding-bottom: 64px;
  margin-top: 0;
  border-radius: 0;
}

@media screen and (min-width: 40em) {
  .emotion-14 {
    margin-top: 32px;
  }
}

@media screen and (min-width: 40em) {
  .emotion-14 {
    border-radius: 32px;
  }
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  padding-left: 64px;
  padding-right: 64px;
  padding-bottom: 32px;
}

@media screen and (min-width: 40em) {
  .emotion-15 {
    padding-left: 128px;
    padding-right: 128px;
  }
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 32px;
  margin-bottom: 16px;
  text-align: center;
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 32px;
  padding-left: 32px;
  padding-right: 32px;
  text-align: center;
  font-size: 16px;
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary);
  background-color: var(--theme-ui-colors-primary);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  padding: 16px;
  width: 100%;
  font-weight: bold;
  border-radius: 16px;
}

.emotion-19:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-19:hover,
.emotion-19:focus {
  background-color: var(--theme-ui-colors-primaryDark);
}

.emotion-19:hover:disabled,
.emotion-19:focus:disabled {
  background-color: var(--theme-ui-colors-primary);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          data-cy="back-button"
          href="/dashboard/admin/subscription"
        >
          <button
            class="emotion-4"
          >
            <div
              class="emotion-5"
            />
          </button>
        </a>
        <div
          class="emotion-6"
        >
          <a
            class="emotion-7"
            href="/dashboard/admin"
          >
            <span
              class="emotion-8"
            >
              Admin
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <a
            class="emotion-7"
            href="/dashboard/admin/subscription"
          >
            <span
              class="emotion-8"
            >
              Plans & Billing
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <div
            class="emotion-13"
          >
            Payment Success
            <span
              class="splitter"
            >
              /
            </span>
          </div>
        </div>
      </div>
      <div
        class="emotion-14"
      >
        <div
          class="emotion-15"
          width="100%"
        />
        <h4
          class="emotion-16"
        >
          Thanks for Subscribing!
        </h4>
        <p
          class="emotion-17"
        >
          Your purchase is complete. A receipt of your order will be sent to you by email. Thanks for supporting us!
        </p>
        <div
          class="emotion-18"
        >
          <a
            href="/dashboard/admin/subscription"
          >
            <button
              class="emotion-19"
            >
              Back to Billing
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
